easybuild/easyconfigs/g/GCCcore/GCCcore-8.1.0.eb

```diff
@@ -0,0 +1,51 @@
+easyblock = 'EB_GCC'
+
+name = 'GCCcore'
+version = '8.1.0'
+
+homepage = 'http://gcc.gnu.org/'
+description = """The GNU Compiler Collection includes front ends for C, C++, Objective-C, Fortran, Java, and Ada,
+ as well as libraries for these languages (libstdc++, libgcj,...)."""
```

Comment:
@boegel Is it worth adding a line in the description about the distinction between `GCC` and `GCCcore`, to make the distinction clearer? It's a question that keeps coming up...

Reply:
@boegel We have HMNS.  What needs testing/checking?

Reply:
Whether we can use `GCC` as a subtoolchain for both `foss` and `intel` (rather than `GCCcore`).

Reply:
This won't work, the reason `GCCcore` exists is because it is *not* classed as a compiler in the hierarchy (i.e., it is not part of the `compiler` family), which allows it to be loaded at the same time as another compiler. It is there to provide a base for all the other compilers.
GCC *is* in the `family` so can't be loaded  at the same time as another compiler. If we did not include it in the family we would not get the desired swapping behaviour. At JSC to avoid confusion we hide `GCCcore`.

Reply:
Folling @ocaisa 's input I did a simple test on our machine

load GCC/6.4.0-2.28
load intel/2018a

I don't see any unloading.  Checking on environment variables, the once related to GCC, eg. EBVERSIONGCC=6.4.0 is still there after loading intel (it is not there when I only load the intel).  A "which gcc" is still pointing at the correct version, so is "mpicc -v".  "mpiicc -v" points to the right intel compiler.

So if I were to go along and build software against this outside EasyBuild this should just work.  Most likely I am still to ignorant on the internal workings of EB.  Could someone clue me where the "explosion" comes?

Reply:
Agreed with @jhein32, I've opened an issue for this: https://github.com/easybuilders/easybuild-easyconfigs/issues/6366

Let's get this one merged @migueldiascosta?